### PR TITLE
New test framework

### PR DESCRIPTION
**Pull request by isaacdorfman**

### CODE REVIEW

File: provider/cluster_resource_test.go [hunk 1]
```diff
@@ -17,295 +17,155 @@ limitations under the License.
...
-		terraform.Source(`
-		  resource "ocm_cluster" "my_cluster" {
-			product		   = "osd"
-		    name           = "my-cluster"
-		    cloud_provider = "aws"
-		    cloud_region   = "us-west-1"
-		  }
-		`)
-		Expect(terraform.Apply()).ToNot(BeZero())
+	It("populates state correctly", func() {
```
Comment (bardielle) [inline provider/cluster_resource_test.go:85]: Add more tests 
Wait & ready
wait & error
wait & not ready

Comment (bardielle) [inline provider/cluster_resource_test.go:85]: Add UT to delete

File: provider/cluster_resource_test.go [hunk 2]
```diff
@@ -17,295 +17,155 @@ limitations under the License.
...
+		availabilityZone := "us-east-1a"
+		ccsEnabled := true
+		awsAccountID := "123456789012"
+		awsAccessKeyID := "AKIAIOSFODNN7EXAMPLE"
+		awsSecretAccessKey := "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+		privateLink := false
+
+		// Create a cluster object
+
+		clusterJson := map[string]interface{}{
```
Comment (bardielle) [inline provider/cluster_resource_test.go:105]: extract to common object

File: provider/cluster_resource_test.go (hunk 1 above)
Comment (isaacdorfman) [reply provider/cluster_resource_test.go]: Most of the logic in delete is about polling the backend for completion and we decided to postpone testing it.

File: provider/cluster_resource_test.go (hunk 2 above)
Comment (isaacdorfman) [reply provider/cluster_resource_test.go:104]: Done

File: provider/cluster_resource_test.go (hunk 1 above)
Comment (isaacdorfman) [reply provider/cluster_resource_test.go]: We decided to postpone testing it.

**Review by bardielle — COMMENTED** [review]
/lgtm

**Review by nirarg — APPROVED** [review]

**Pull request merged**